src/System.Net.Http/src/System/Net/Http/HttpClientHandler.Unix.cs

```diff
@@ -160,6 +161,10 @@ protected override void Dispose(bool disposing)
 
         protected internal override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         {
+            if (DiagnosticsHandler.IsEnabled())
```

Comment:
Could add `IsEnabled()` as non-virtual call to `DiagnosticsListener` and have it return `diagnosticSubscription != null`?